invenio/modules/deposit/testsuite/test_deposit_models.py

```diff
@@ -57,9 +63,13 @@ def test_create(self):
         d = Deposition.create(user)
         assert d.type == self.DefaultType
         assert Deposition.get(d.id).type == self.DefaultType
-        d = Deposition.create(user, type=self.AnotherType)
-        assert d.type == self.AnotherType
-        assert Deposition.get(d.id).type == self.AnotherType
+        d2 = Deposition.create(user, type=self.AnotherType)
+        assert d2.type == self.AnotherType
+        assert Deposition.get(d2.id).type == self.AnotherType
+
+        # remove the records
+        Deposition.delete(d)
```

Comment:
Deposition is just a wrapper a around a ``BibWorkflowObject`` (workflows module) so you should test something like:

```python
bwl_d2_count = BibWorkflowObject.query.filter_by(uuid=d2.workflow_object.id_workflow).count()
d.delete()
assert Workflow.query.filter_by(uuid=d.workflow_object.id_workflow).count() == 0
assert  BibWorkflowObject.query.filter_by(id_workflow=d.workflow_object.id_workflow).count() == 0
```
and that no other objects where deleted, e.g.:

```python
assert Workflow.query.filter_by(uuid=d2.workflow_object.id_workflow).count() == 1
assert BibWorkflowObject.query.filter_by(uuid=d2.workflow_object.id_workflow).count() == bwl_d2_count
#...
d2.delete()
```